src/main/java/org/sagebionetworks/bridge/sdk/models/UploadRequest.java

```
@@ -89,4 +77,94 @@ public String toString() {
+        public Builder withName(String name) {
+            this.name = name;
+            return this;
+        }
+
+        /** @see org.sagebionetworks.bridge.sdk.models.UploadRequest#getContentLength */
+        public long getContentLength() {
+            return contentLength;
+        }
+
+        /** @see org.sagebionetworks.bridge.sdk.models.UploadRequest#getContentLength */
+        public Builder withContentLength(long contentLength) {
+            this.contentLength = contentLength;
+            return this;
+        }
+
+        /** @see org.sagebionetworks.bridge.sdk.models.UploadRequest#getContentMd5 */
+        public String getContentMd5() {
+            return contentMd5;
+        }
+
+        /** @see org.sagebionetworks.bridge.sdk.models.UploadRequest#getContentMd5 */
+        public Builder withContentMd5(String contentMd5) {
+            this.contentMd5 = contentMd5;
+            return this;
+        }
+
+        /** @see org.sagebionetworks.bridge.sdk.models.UploadRequest#getContentType */
+        public String getContentType() {
+            return contentType;
+        }
+
+        /** @see org.sagebionetworks.bridge.sdk.models.UploadRequest#getContentType */
+        public Builder withContentType(String contentType) {
+            this.contentType = contentType;
+            return this;
+        }
+
+        /**
+         * Builds and validates an UploadRequest. This validates that name is non-null and non-empty, content length is
+         * non-negative, content MD5 is non-null and non-empty, and content type is non-null and non-empty. This method
+         * does not validate that the named file exists or that its length and MD5 are accurate. To ensure account
+         * length and MD5, use {@link #withFile}.
+         *
+         * @return validated UploadRequest
+         * @throws BridgeSDKException
+         *         if called with invalid fields
+         */
+        public UploadRequest build() throws BridgeSDKException {
```

I think this should throw InvalidEntityException, to emphasize the SDK users has constructed something wrong, not that the SDK has an error. Same with validation in the other builder.